### PR TITLE
[Frontend] Add Scan Page

### LINKED ISSUE #15: [Frontend] Add Scan Page
- Access Camera
- Display live camera feed
- Scan live camera feed for QR Codes

[This package](https://github.com/nimiq/qr-scanner) handles those three things already

- read data from QR Code (should contain userid and token, optionally contains name, email, phone)

- fire http request to server to verify userid and token
- pop up a toast/modal showing whether the qrcode was verified as legit
- if it matches, add name, email, phone, current time to localStorage

### PR DESCRIPTION
**Pull request by daniel-weisdorf**
Opens a video display - you can click scan to attempt to scan

Toasts if it 
- doesnt find QR code
- if QR code is in wrong format
- if QR code is in right format but is invalid
- if QR code is valid

If you want to test it yourself, open up chrome console and do
`localStorage.setItem('userId', 'a-b-c-d-id')` and `localStorage.setItem('totpSecret', 'id')`

Then open the page twice - go to "individual profile" on one, and "scan" on the other. Point your webcam at your "individual profile" qr code and it should work. if you `localStorage.setItem('totpSecret', '{anything else}')` it should start to fail

Closes #15

### CODE REVIEW

File: frontend/src/pages/Scan.js [hunk 1]
```diff
@@ -1,15 +1,108 @@
...
+    async processQrCodeScan(qrCode) {
+        console.log(qrCode);
+
+        try {
+            const fields = qrCode.split(",");
+            const userId = fields[0];
+            const token = fields[1];
+            const name = fields[2];
+            const email = fields[3];
+            const phone = fields[4];
```
Comment (ahmedadan) [inline frontend/src/pages/Scan.js:36]: Will this throw an exception if the name, email, phone are not provided?

**Review by ahmedadan — APPROVED** [review]
LGTM, FYI code reviews on mobile might as well be torture. One outstanding question re: optional values.

File: frontend/src/pages/Scan.js (hunk 1 above)
Comment (daniel-weisdorf) [reply frontend/src/pages/Scan.js:36]: if my implementation of the QR code was good, potentially, but right now I'm putting `null` in the QR code if the value doesn't exist, so nothing explodes

Comment (daniel-weisdorf) [reply frontend/src/pages/Scan.js:36]: ive been testing with just userid and token so it SHOULD only break with a malformed qr code, in which case throwing the exception is desired

**Pull request merged**